torch/_inductor/fx_passes/post_grad.py

```diff
@@ -1544,6 +1544,62 @@ def repl(inp, x1, x2, alpha, beta):
     match.replace_by_example(repl, [inp, mat1, mat2, alpha, beta])
 
 
+def is_valid_addmm_activation_fusion(match: Match) -> bool:
+    if not is_gpu(match.kwargs["inp"].meta["val"].device.type):
+        return False
+
+    # Only beta == 1 implies activation epilogue in addmm
+    if match.kwargs["beta"] not in (1, 1.0, 1+0j, 1-0j):
+        return False
+
+    # GELU epilogue fusion only for the "tanh" approximation
+    if "approximate" in match.kwargs:
+        # Somehow match.kwargs.setdefault("approximate", <default>) does not work
+        gelu_approximation = match.kwargs["approximate"]
+        if gelu_approximation != "tanh":
+            return False
+
+    return not has_uses_tagged_as(
+        match.output_node(),
+        (torch.Tag.pointwise, torch.Tag.reduction),
+    )
+
+
+@register_graph_pattern(
+    CallFunction(
+        aten.relu,
+        CallFunction(aten.addmm, KeywordArg("inp"), Arg(), Arg(), beta=KeywordArg("beta"), alpha=KeywordArg("alpha")),
+    ),
+    # pyrefly: ignore [bad-argument-type]
+    pass_dict=pass_patterns[1],
+    extra_check=is_valid_addmm_activation_fusion,
+)
+def relu_addmm_fusion(match: Match, mat1, mat2, *, inp, beta, alpha):
+    def replacement(inp, mat1, mat2, beta, alpha):
+        return aten._addmm_activation(inp, mat1, mat2, beta=beta, alpha=alpha)
+
+    # pyrefly: ignore [bad-argument-type]
+    match.replace_by_example(replacement, [inp, mat1, mat2, beta, alpha])
+
+
+@register_graph_pattern(
+    CallFunction(
+        aten.gelu,
+        CallFunction(aten.addmm, KeywordArg("inp"), Arg(), Arg(), beta=KeywordArg("beta"), alpha=KeywordArg("alpha")),
+        KeywordArg("approximate"),
+    ),
```

Comment:
This pattern is not being captured because `gelu` is being decomposed. What could be an alternative? Or is it a bug? CC @eellison